index.html

```diff
@@ -609,106 +609,211 @@
+navigator.nfc.watch((message) => {
+  console.log('NDEF message received from URL ' + message.url);
 
-      // Specify custom format, to control NDEF data layout.
-      var message = {
-        url: "https://company.com/path/game/update",  // app specific path
-        data: [{
-          recordType: "json",
-          mediaType: "application/json",
-          data: { level: 3, points: 4500, lives: 3 }
-        }]
-      };
+  if (message.data[0].recordType == 'empty') {
+    navigator.nfc.push({
+      url: "/mypath/myapp",  // path set by the application
+      data: [{
+        recordType: "url",
+        data: "Initial data"
+      }]
+    });
+  }
+  processMessage(message);
+}).then(() => {
+  console.log("Watch added.");
+}).catch((error) => {
+  console.log("Adding watch failed: " + error.name)
+});
+
+function processMessage(message) {
+  message.data.forEach((record) => {
+    if (record.recordType == "string") {
+      console.log('Data is string: ' + record.data);
+    } else if (record.recordType == 'json') {
+      processJSON(record.data);
+    } else if (record.recordType == 'url') {
+    console.log('Data is URL: ' + record.data);
+    } else if (record.recordType == 'opaque') {
+      processBinary(record.data);
+  });
+};
+
+function processBinary(data) {
+  console.log('Known (string) binary data: ');
+  console.log(String.fromCharCode.apply(null, new Uint16Array(data)));
+};
+
+function processJSON(data) {
+  var obj = JSON.parse(data);
+  console.log('JSON data: ' + obj.myProperty.toString());
+};
+    </pre>
+  </aside>
+
+  <aside title="Save and restore game progress with another device"
+       class="example">
+    <p>
+      Filtering of relevant data sources can be done by the use of
+      the <a>NFCWatchOptions</a>. Below we accept any URL with <code>
+      "/mypath/mygame/"</code> in its path. When we read the data,
+      we immediately update the game progress by issueing a push
+      with a custom NDEF data layout.
+    </p>
+    <p>
+      The example allows reading and pushing to both peers and tags,
+      whichever one is tapped first.
+    </p>
+    <pre class="highlight">
+navigator.nfc.watch(reader, { url: "*/mypath/mygame/*" });
+
+function reader(message) {
+  console.log("Source:     " + message.url);
+  console.log("Game state: " + message.data);
+
+  var message = {
+    url: "https://company.com/path/game/update",  // app specific path
+    data: [{
+      recordType: "json",
+      mediaType: "application/json",
+      data: { level: 3, points: 4500, lives: 3 }
+    }]
+  };
+
+  navigator.nfc.push(message).then(() => {
+    console.log('Progress stored!');
+  }).catch((error) => {
+    console.log('Failure, please try again.');
+  });
+};
+    </pre>
+  </aside>
+
+  <aside title="Push and read JSON (auto serialized and deserialized)"
+       class="example">
+    <p>
+      Storing and receiving JSON data is easy as serialization and deserialization
+      happens automatically.
```

Comment:
Then we need to remove "object" type from NFCRecordData. At the moment spec implies that we have automatic serialization / deserialization for "objects".

We can also change "object" to "Dictionary" to reduce scope of object. Then it will be allowed to pass associative arrays. However, (de)serialization will be needed in any case.